src/Components/Aspire.Azure.AI.OpenAI/AzureOpenAISettings.cs

```diff
@@ -49,23 +49,26 @@ public sealed class AzureOpenAISettings : IConnectionStringSettings
 
     void IConnectionStringSettings.ParseConnectionString(string? connectionString)
     {
-        var connectionBuilder = new DbConnectionStringBuilder
-        {
-            ConnectionString = connectionString
-        };
-
-        if (connectionBuilder.ContainsKey(ConnectionStringEndpoint) && Uri.TryCreate(connectionBuilder[ConnectionStringEndpoint].ToString(), UriKind.Absolute, out var serviceUri))
-        {
-            Endpoint = serviceUri;
-        }
-        else if (Uri.TryCreate(connectionString, UriKind.Absolute, out var uri))
+        if (Uri.TryCreate(connectionString, UriKind.Absolute, out var uri))
         {
             Endpoint = uri;
         }
-
-        if (connectionBuilder.ContainsKey(ConnectionStringKey))
+        else
         {
-            Key = connectionBuilder[ConnectionStringKey].ToString();
+            var connectionBuilder = new DbConnectionStringBuilder
+            {
+                ConnectionString = connectionString
+            };
+
+            if (connectionBuilder.ContainsKey(ConnectionStringEndpoint) && Uri.TryCreate(connectionBuilder[ConnectionStringEndpoint].ToString(), UriKind.Absolute, out var serviceUri))
```

Comment:
Wouldn't this still fail if my connectionString value was "Endpoint=http://foobar.com:12345;Key=abc123"

Reply:
I don't think, what do you have in mind? The current tests use `"Endpoint=https://aspireopenaitests.openai.azure.com/;Key=fake"`

I added some more tests to show you.